core/ecschema-editing/src/Editing/RelationshipClasses.ts

```diff
@@ -115,7 +115,7 @@ export class RelationshipClasses extends ECClasses {
     try {
       await this.validate(relClass!);
     } catch(e: any) {
-      relClass!.baseClass = baseClass;
+      await relClass!.setBaseClass(baseClass);
```

Comment:
Should we add some check here (or actually in line 111) to ensure that relClass is not undefined? Since it's a lookup by key, I assume it could be happen (eg when loading from a referenced schema) or did I missed some previous validation here?

Reply:
I added a catch block on line 112.